gips/0083.md

```diff
@@ -0,0 +1,250 @@
+
+The Payment Gateway CAN have systems to perform health checks (e.g. on `/health` or `/info` endpoints), or other checks to ensure active, up-to-date and properly configured providers are offered to end users. A Payment Gateway CAN check block height to ensure the backing provider is close to chain head before offering it to users.
+
+
+### Tools
+
+THIS SECTION IS NON-NORMATIVE. Different tools can exist to do those operations. We are listing some for informational purposes here.
+
+For Indexers, in order to register a new Substreams provider on-chain, and show up on the Payment Gateway, you can use commands such as:
+
+```
+substreams network register --operator-wallet 0x123123123 --operator-priv-key-file my.key --service substreams-v1 --endpoint mainnet.eth.example.com --provider-logo https://www.example.com/static/mylogo.png --provider-id example-company --provider-name "Example Company" --network ethereum
+```
+
+This command allows providers to register their service, specifying their operator wallet, private key file, the service type (`substreams-v1`), the endpoint URL, logo, name, and the supported network. The registration data is signed by the operator's key.  An update to an existing registration can be achieved by re-running the command with updated parameters, based on the `service`, `providerId` and `network` tripled.  A corresponding `unregister` or `revoke` command would be added to allow providers to take down their service from the front-end.
+
+### References
+
+ - [DataEdge addresses](https://github.com/graphprotocol/contracts/blob/main/packages/data-edge/addresses.js)
+- The relevant contract sources are `DataEdge` ([DataEdge.sol](https://github.com/graphprotocol/contracts/blob/main/packages/data-edge/contracts/DataEdge.sol)) and `EventfulDataEdge` ([EventfulDataEdge.sol](https://github.com/graphprotocol/contracts/blob/main/packages/data-edge/contracts/EventfulDataEdge.sol)). `EventfulDataEdge` emits an event upon registration, while `DataEdge` does not.
+
+# Meaning and commitment
+
+Approving this GIP settles affirmatively the question of whether Substreams is "On The Network".
+
+# Dependencies & Backwards Compatibility
+
+The only dependency for this GIP is the ratification of the changes to GIP-0009 as proposed in this Pull Request.
+
+Most of the elements described above are already rolled out and currently work.
+
+To account for signed attestation, some backwards compatible fields were be added to the Substreams requests and responses.  These fields should not affect current Substreams rollout, but will be necessary for those wanting to join the Network.
+
+
+# Risks and Security Considerations
+
+Risks associated with data integrity and malicious actors will be mitigated through signed attestations, the arbitration process, and economic security enforced via slashing.
```

Comment:
I think it's worth noting there is some centralization risk in the payment gateway until TAP is integrated. (Indexers need to trust the gateway for payment, so it'd be hard for a new payment gateway to spin up and use substreams on the network). And in the same vein, there is some counterparty risk for indexers, as they have to trust that the payment gateway will pay - this can be mitigated by requesting payment early and often.

Reply:
ok, adding those notes